### PR TITLE
refactor(motion_utils): clear the repeat definitions and correct the dependencies 

### PR DESCRIPTION
**Pull request by HansOersted**
## Description

A function in one .cpp file can be used in others. A clean dependency should be in the following scenario:

Functions that are commonly called are encouraged to be aggregated in an independent location, e.g., .cpp file.

The current program lacks a clean dependency. Therefore, the primary aim of this ticket is to clean the dependency by collecting the common functions.

 
Specifically, move the relevant functions in 

`autoware.universe/planning/behavior_velocity_planner_common/src/utilization/trajectory_utils.cpp`

`autoware.universe/planning/behavior_path_planner_common/src/utils/path_utils.cpp`

to

`autoware.universe/common/motion_utils/include/motion_utils/trajectory/conversion.hpp`

Further, lerpOrientation() is also collected in the 
`common/interpolation/src/spherical_linear_interpolation.cpp`

## Related links

https://tier4.atlassian.net/browse/RT1-4484

## Tests performed

colcon build --cmake-args -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_EXPORT_COMPILE_COMMANDS=1 --symlink-install --continue-on-error

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [X] I've confirmed the [contribution guidelines].
- [X] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: common/motion_utils/src/trajectory/conversion.cpp [hunk 1]
```diff
@@ -12,12 +12,19 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include "motion_utils/trajectory/tmp_conversion.hpp"
+#include "motion_utils/trajectory/conversion.hpp"
+
+#include "autoware_auto_planning_msgs/msg/detail/path__struct.hpp"
+#include "autoware_auto_planning_msgs/msg/detail/path_point_with_lane_id__struct.hpp"
+#include "autoware_auto_planning_msgs/msg/detail/path_with_lane_id__struct.hpp"
+#include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>
```
Comment (zulfaqar-azmi-t4) [inline common/motion_utils/src/trajectory/conversion.cpp:20]: I have a feeling this header is not needed, but I might be wrong.

File: planning/behavior_velocity_planner_common/src/utilization/trajectory_utils.cpp [hunk 2]
```diff
@@ -13,6 +13,8 @@
 // limitations under the License.
 
 // #include <behavior_velocity_planner_common/utilization/boost_geometry_helper.hpp>
+#include "../../../../common/motion_utils/include/motion_utils/trajectory/conversion.hpp"
```
Comment (isamu-takagi) [inline planning/behavior_velocity_planner_common/src/utilization/trajectory_utils.cpp:16]:
```suggestion
#include <motion_utils/trajectory/conversion.hpp>
```

File: common/motion_utils/src/trajectory/conversion.cpp (hunk 1 above)
Comment (HansOersted) [reply common/motion_utils/src/trajectory/conversion.cpp]: Indeed. Unnecessary dependencies deleted in the new commit [https://github.com/autowarefoundation/autoware.universe/pull/5909/commits/499cd278476678fb7fc79e4acc7f146c4f422da6](https://github.com/autowarefoundation/autoware.universe/pull/5909/commits/499cd278476678fb7fc79e4acc7f146c4f422da6)

1. [deleted] `#include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>`
for `lerpOrientation()`, which was moved to another place.





2. `#include "autoware_auto_planning_msgs/msg/detail/path__struct.hpp"` and 
`#include "autoware_auto_planning_msgs/msg/detail/path_point_with_lane_id__struct.hpp"` and 
`#include "autoware_auto_planning_msgs/msg/detail/path_with_lane_id__struct.hpp"` 
for 
`Path`
`PathPointWithLaneId`
`PathWithLaneId`

**Comment by satoshi-ota** [comment]
@HansOersted Could you fix conflicts...?

**Review by kyoichi-sugahara — APPROVED** [review]
LGTM!!!

**Review by satoshi-ota — APPROVED** [review]

**Review by isamu-takagi — APPROVED** [review]

**Review by tkimura4 — APPROVED** [review]

**Review by takayuki5168 — APPROVED** [review]

**Pull request merged**